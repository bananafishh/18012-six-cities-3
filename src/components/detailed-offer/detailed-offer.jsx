import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {
  getReviews,
  getNearbyOffers,
  getReviewPostingStatus,
  getReviewPostingError,
  getOffer,
} from '../../reducer/data/selector';

import {Operation} from '../../reducer/data/data';
import withReviewFieldsChange from '../../hocs/with-review-fields-change/with-review-fields-change';

import {
  Offer,
  PlaceType,
  AuthStatus,
  AppRoute,
} from '../../constants';

import {getRatingInPercent, pluralizeWord} from '../../utils';

import ReviewsList from '../reviews-list/reviews-list.jsx';
import OffersList from '../offers-list/offers-list.jsx';
import Map from '../map/map.jsx';
import ReviewForm from '../review-form/review-form.jsx';

const ReviewFormWrapped = withReviewFieldsChange(ReviewForm);

class DetailedOffer extends PureComponent {
  componentDidMount() {
    const {
      id,
      onOfferDataLoad,
    } = this.props;

    onOfferDataLoad(id);
    window.scrollTo(0, 0);
  }

  componentDidUpdate(prevProps) {
    const {
      id,
      onOfferDataLoad,
    } = this.props;

    if (id !== prevProps.id) {
      onOfferDataLoad(id);
      window.scrollTo(0, 0);
    }
  }

  render() {
    const {
      offer,
      reviews,
      nearbyOffers,
      authStatus,
      isReviewPosting,
      isReviewPostingError,
      onReviewSend,
      onBookmarkClick,
      history,
    } = this.props;

    const {
      id,
      title,
      type,
      price,
      images,
      rating,
      isPremium,
      isFavorite,
      description,
      bedrooms,
      maxAdults,
      goods,
      host: {
        name,
        avatarUrl,
        isPro,
      },
    } = offer;

    const handleBookmarkClick = (offerId, isOfferFavorite) => (
      authStatus === AuthStatus.AUTH
        ? onBookmarkClick(offerId, isOfferFavorite)
        : history.push(AppRoute.SIGN_IN)
    );

    const sortedReviews = reviews.length
      ? reviews
        .slice(0, Offer.REVIEWS_MAX)
        .sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime())
      : [];

    const nearbyOffersForRender = nearbyOffers.slice(0, Offer.NEARBY_OFFERS_MAX);
    const offers = [offer, ...nearbyOffersForRender];

    return (
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {!!images.length && images
                .slice(0, Offer.IMAGES_MAX)
                .map((picture) => (
                  <div key={picture} className="property__image-wrapper">
                    <img className="property__image" src={picture} alt="Photo studio"/>
                  </div>
                ))}
            </div>
          </div>

          <div className="property__container container">
            <div className="property__wrapper">
              {isPremium && (
                <div className="property__mark">
                  <span>Premium</span>
                </div>
              )}

              <div className="property__name-wrapper">
                <h1 className="property__name">{title}</h1>

                <button
                  className={`property__bookmark-button${isFavorite ? ` property__bookmark-button--active` : ``} button`}
                  type="button"
                  onClick={() => handleBookmarkClick(id, isFavorite)}
                >
                  <svg className="property__bookmark-icon" width="31" height="33">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>

                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>

              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: `${getRatingInPercent(rating)}%`}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>

                <span className="property__rating-value rating__value">{rating}</span>
              </div>

              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {PlaceType[type.toUpperCase()]}
                </li>

                <li className="property__feature property__feature--bedrooms">
                  {bedrooms} {pluralizeWord(`Bedroom`, bedrooms)}
                </li>

                <li className="property__feature property__feature--adults">
                  Max {maxAdults} {pluralizeWord(`adult`, maxAdults)}
                </li>
              </ul>

              <div className="property__price">
                <b className="property__price-value">&euro;{price}</b>
                {` `}
                <span className="property__price-text">&nbsp;night</span>
              </div>

              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>

                <ul className="property__inside-list">
                  {!!goods.length && goods.map((householdItem) => (
                    <li key={householdItem} className="property__inside-item">{householdItem}</li>
                  ))}
                </ul>
              </div>

              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>

                <div className="property__host-user user">
                  <div
                    className={`property__avatar-wrapper${isPro ? ` property__avatar-wrapper--pro` : ``} user__avatar-wrapper`}>
                    <img
                      className="property__avatar user__avatar"
                      src={`/${avatarUrl}`}
                      width="74"
                      height="74"
                      alt="Host avatar"
                    />
                  </div>

                  <span className="property__user-name">{name}</span>
                </div>

                <div className="property__description">
                  {description
                    .split(`\n`)
                    .map((descriptionItem, index) => (
                      <p key={index} className="property__text">{descriptionItem}</p>
                    ))
                  }
                </div>
              </div>

              <section className="property__reviews reviews">
                <h2 className="reviews__title">Reviews &middot; <span
                  className="reviews__amount">{sortedReviews.length}</span></h2>

                <ReviewsList reviews={sortedReviews}/>

                {authStatus === AuthStatus.AUTH && (
                  <ReviewFormWrapped
                    offerId={id}
                    isReviewPosting={isReviewPosting}
                    isReviewPostingError={isReviewPostingError}
                    onReviewSend={onReviewSend}
                  />
                )}
              </section>
            </div>
          </div>

          <section className="property__map map">
            <Map
              offers={offers}
              activeOfferId={id}
            />
          </section>
        </section>

        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>

            <div className="near-places__list places__list">
              <OffersList
                offerMix={{
                  card: `near-places__card`,
                  img: `cities__image-wrapper`,
                }}
                imgSize={{
                  width: `260`,
                  height: `200`,
                }}
                offers={nearbyOffersForRender}
                history={history}
                authStatus={authStatus}
                onBookmarkClick={onBookmarkClick}
              />
            </div>
          </section>
        </div>
      </main>
    );
  }
}

DetailedOffer.defaultProps = {
  offer: {},
  images: [],
  goods: [],
  reviews: [],
};

DetailedOffer.propTypes = {
  id: PropTypes.number.isRequired,
  offer: PropTypes.shape({
    id: PropTypes.number,
    city: PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.shape({
        latitude: PropTypes.number,
        longitude: PropTypes.number,
        zoom: PropTypes.number,
      }),
    }),
    title: PropTypes.string,
    type: PropTypes.string,
    price: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    rating: PropTypes.number,
    isPremium: PropTypes.bool,
    isFavorite: PropTypes.bool,
    description: PropTypes.string,
    bedrooms: PropTypes.number,
    maxAdults: PropTypes.number,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      name: PropTypes.string,
      isPro: PropTypes.bool,
      avatarUrl: PropTypes.string,
    }),
  }).isRequired,
  reviews: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    text: PropTypes.string,
    rating: PropTypes.number,
    date: PropTypes.string,
    user: PropTypes.shape({
      name: PropTypes.string,
      avatarUrl: PropTypes.string,
    })
  })).isRequired,
  nearbyOffers: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    type: PropTypes.string,
    price: PropTypes.number,
    previewImage: PropTypes.string,
    rating: PropTypes.number,
    isPremium: PropTypes.bool,
    isFavorite: PropTypes.bool,
  })).isRequired,
  authStatus: PropTypes.string.isRequired,
  isReviewPosting: PropTypes.bool.isRequired,
  isReviewPostingError: PropTypes.bool.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
  onOfferDataLoad: PropTypes.func.isRequired,
  onReviewSend: PropTypes.func.isRequired,
  onBookmarkClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => ({
  offer: getOffer(state, ownProps),
  reviews: getReviews(state),
  nearbyOffers: getNearbyOffers(state),
  isReviewPosting: getReviewPostingStatus(state),
  isReviewPostingError: getReviewPostingError(state),
});

const mapDispatchToProps = (dispatch) => ({
  onOfferDataLoad(offerId) {
    dispatch(Operation.loadReviews(offerId));
    dispatch(Operation.loadNearbyOffers(offerId));
  },
  onReviewSend(offerId, review) {
    const postReviewPromise = dispatch(Operation.postReview(offerId, review));

    return postReviewPromise;
  },
  onBookmarkClick(offerId, isFavorite) {
    dispatch(Operation.toggleFavorite(offerId, isFavorite));
  },
});

export {DetailedOffer};
export default connect(mapStateToProps, mapDispatchToProps)(DetailedOffer);
