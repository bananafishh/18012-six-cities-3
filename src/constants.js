export const RATING_MAX = 5;

export const REVIEWS_ON_PAGE_MAX = 10;

export const NEARBY_OFFERS_MAX = 3;

export const OFFER_IMAGES_MAX = 6;

export const PlaceType = {
  APARTMENT: `Apartment`,
  ROOM: `Private Room`,
  HOUSE: `House`,
  HOTEL: `Hotel`,
};

export const MONTHS = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`,
];

export const DateFormat = {
  FULL: `MMMM DD, YYYY`,
  WITH_DASHES: `YYYY-MM-DD`,
};

export const SortingOption = {
  POPULAR: `popular`,
  PRICE_LOW_TO_HIGH: `priceLowToHigh`,
  PRICE_HIGH_TO_LOW: `priceHighToLow`,
  TOP_RATED_FIRST: `topRatedFirst`,
};

export const SORTING_OPTIONS = [
  {
    label: `Popular`,
    value: SortingOption.POPULAR,
  },
  {
    label: `Price: low to high`,
    value: SortingOption.PRICE_LOW_TO_HIGH,
  },
  {
    label: `Price: high to low`,
    value: SortingOption.PRICE_HIGH_TO_LOW,
  },
  {
    label: `Top rated first`,
    value: SortingOption.TOP_RATED_FIRST,
  },
];

export const AuthorizationStatus = {
  AUTHORIZED: `AUTHORIZED`,
  UNAUTHORIZED: `UNAUTHORIZED`,
};
