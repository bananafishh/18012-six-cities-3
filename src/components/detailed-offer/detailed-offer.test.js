import React from 'react';
import renderer from 'react-test-renderer';

import {AuthStatus} from '../../constants';

import {DetailedOffer} from './detailed-offer';

const offers = [
  {
    id: 1,
    title: `Luxe 1-Bedroom Flat Near Manhattan`,
    type: `apartment`,
    price: 120,
    previewImage: `https://placeimg.com/260/200/arch/1`,
    rating: 4.5,
    isPremium: true,
    isFavorite: false,
  },
  {
    id: 2,
    title: `Bright & Airy in Highland Park`,
    type: `house`,
    price: 200,
    previewImage: `https://placeimg.com/260/200/arch/2`,
    rating: 3.9,
    isPremium: false,
    isFavorite: true,
  },
];

const offer = {
  id: 1,
  city: {
    name: `Dusseldorf`,
    location: {
      latitude: 51.225402,
      longitude: 6.776314,
      zoom: 13,
    },
  },
  title: `Luxe 1-Bedroom Flat Near Manhattan`,
  type: `apartment`,
  price: 120,
  images: [
    `https://placeimg.com/260/200/arch/1`,
    `https://placeimg.com/260/200/arch/2`,
    `https://placeimg.com/260/200/arch/3`,
    `https://placeimg.com/260/200/arch/4`,
    `https://placeimg.com/260/200/arch/5`,
    `https://placeimg.com/260/200/arch/6`,
  ],
  rating: 4.5,
  isPremium: true,
  isFavorite: false,
  description: `This is a unique and lovely apartment located in the heart of the Manhattan. Its truly original interior design will help you to experience New-York's unforgettable charm.
  The apartment is very clean. It has high ceilings, TV set with a selection of movies to enjoy on a rainy day, hi-fi system, high speed wi-fi and a tiny but fully-equipped kitchenette.`,
  bedrooms: 1,
  maxAdults: 2,
  goods: [
    `Wi-Fi`,
    `Washing machine`,
    `Towels`,
    `Heating`,
    `Coffee machine`,
    `Kitchen`,
    `Dishwasher`,
    `Cabel TV`,
    `Fridge`,
  ],
  host: {
    avatarUrl: `https://api.adorable.io/avatars/128/1`,
    name: `Abott`,
    isPro: true,
  },
};

const reviews = [
  {
    id: 1,
    text: `Excellent location in a charming building. It’s about the size of a typical hotel room and 
      perhaps a little smaller than Airbnb places tend to be overall. Would recommend though as it’s 
      a great value and location.`,
    rating: 5,
    date: `2019-04-03T14:13:56.569Z`,
    user: {
      name: `Steven`,
      picture: `https://api.adorable.io/avatars/128/1`,
    },
  },
  {
    id: 2,
    text: `Apartment was very pretty and quaint! Location was excellent , very close to the main 
      sights in Porto. Abott greeted us and was very helpful, gave us some great tips.`,
    rating: 4,
    date: `2019-05-08T14:13:56.569Z`,
    user: {
      name: `Monica`,
      picture: `https://api.adorable.io/avatars/128/2`,
    },
  },
];

const isReviewPosting = false;

const isReviewPostingError = false;

it(`Компонент «DetailedOffer» рендерится корректно`, () => {
  const tree = renderer
    .create(
        <DetailedOffer
          offer={offer}
          reviews={reviews}
          nearbyOffers={offers}
          authStatus={AuthStatus.AUTH}
          isReviewPosting={isReviewPosting}
          isReviewPostingError={isReviewPostingError}
          onNearbyOfferTitleClick={() => {}}
          onOfferDataLoad={() => {}}
          onReviewSend={() => {}}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
