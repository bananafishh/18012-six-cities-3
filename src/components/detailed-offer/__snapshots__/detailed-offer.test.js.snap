// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент «DetailedOffer» рендерится корректно 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://placeimg.com/260/200/arch/1"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://placeimg.com/260/200/arch/2"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://placeimg.com/260/200/arch/3"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://placeimg.com/260/200/arch/4"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://placeimg.com/260/200/arch/5"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://placeimg.com/260/200/arch/6"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Luxe 1-Bedroom Flat Near Manhattan
          </h1>
          <button
            className="property__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "90%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            4.5
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            1
             
            Bedroom
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            2
             
            adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            120
          </b>
           
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Wi-Fi
            </li>
            <li
              className="property__inside-item"
            >
              Washing machine
            </li>
            <li
              className="property__inside-item"
            >
              Towels
            </li>
            <li
              className="property__inside-item"
            >
              Heating
            </li>
            <li
              className="property__inside-item"
            >
              Coffee machine
            </li>
            <li
              className="property__inside-item"
            >
              Kitchen
            </li>
            <li
              className="property__inside-item"
            >
              Dishwasher
            </li>
            <li
              className="property__inside-item"
            >
              Cabel TV
            </li>
            <li
              className="property__inside-item"
            >
              Fridge
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="/https://api.adorable.io/avatars/128/1"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Abott
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              This is a unique and lovely apartment located in the heart of the Manhattan. Its truly original interior design will help you to experience New-York's unforgettable charm.
            </p>
            <p
              className="property__text"
            >
                The apartment is very clean. It has high ceilings, TV set with a selection of movies to enjoy on a rainy day, hi-fi system, high speed wi-fi and a tiny but fully-equipped kitchenette.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews · 
            <span
              className="reviews__amount"
            >
              2
            </span>
          </h2>
          <ul
            className="reviews__list"
          >
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviewer's avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Monica
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  Apartment was very pretty and quaint! Location was excellent , very close to the main 
      sights in Porto. Abott greeted us and was very helpful, gave us some great tips.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-05-8"
                >
                  May 08, 2019
                </time>
              </div>
            </li>
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviewer's avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Steven
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  Excellent location in a charming building. It’s about the size of a typical hotel room and 
      perhaps a little smaller than Airbnb places tend to be overall. Would recommend though as it’s 
      a great value and location.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-04-3"
                >
                  April 03, 2019
                </time>
              </div>
            </li>
          </ul>
          <form
            action="#"
            className="reviews__form form"
            method="post"
            onSubmit={[Function]}
          >
            <label
              className="reviews__label form__label"
              htmlFor="review"
            >
              Your review
            </label>
            <div
              className="reviews__rating-form form__rating"
            >
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                disabled={false}
                id="5-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={5}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="5-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                disabled={false}
                id="4-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={4}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="4-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                disabled={false}
                id="3-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={3}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="3-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                disabled={false}
                id="2-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={2}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="2-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                disabled={false}
                id="1-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={1}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="1-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
            </div>
            <textarea
              className="reviews__textarea form__textarea"
              disabled={false}
              id="review"
              name="review"
              onChange={[Function]}
              placeholder="Tell how was your stay, what you like and what can be improved"
              value=""
            />
            <div
              className="reviews__button-wrapper"
            >
              <p
                className="reviews__help"
              >
                To submit review please make sure to set 
                <span
                  className="reviews__star"
                >
                  rating
                </span>
                 and describe your stay with at least 
                <b
                  className="reviews__text-amount"
                >
                  50 characters
                </b>
                .
              </p>
              <button
                className="reviews__submit form__submit button"
                disabled={true}
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <div
        style={
          Object {
            "height": "100%",
          }
        }
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      >
        <article
          className="place-card near-places__card"
          onMouseEnter={[Function]}
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="place-card__image-wrapper cities__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://placeimg.com/260/200/arch/1"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  120
                </b>
                 
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "90%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/1"
                onClick={[Function]}
              >
                Luxe 1-Bedroom Flat Near Manhattan
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Apartment
            </p>
          </div>
        </article>
        <article
          className="place-card near-places__card"
          onMouseEnter={[Function]}
        >
          <div
            className="place-card__image-wrapper cities__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://placeimg.com/260/200/arch/2"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  200
                </b>
                 
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button place-card__bookmark-button--active button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "78%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/2"
                onClick={[Function]}
              >
                Bright & Airy in Highland Park
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              House
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
